### PR TITLE
add cache for admin/user token;

### PR DESCRIPTION
**Pull request by jesuspg**
@hmunfru @jframos

### CODE REVIEW

**Comment by hmunfru** [comment]
LGTM

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/auth/OpenStackAuthenticationToken.java [hunk 1]
```diff
@@ -109,200 +111,61 @@ public void initialize(ArrayList<Object> params) {
...
-    }
+            Invocation.Builder builder = wr.request(MediaType.APPLICATION_JSON).accept(MediaType.APPLICATION_JSON);
 
-    private ArrayList<Object> executePostRequest(HttpPost postRequest) {
-        HttpResponse response;
-//        httpClient = new DefaultHttpClient();
+            response = builder.post(Entity.entity(payload, MediaType.APPLICATION_JSON));
 
-        ArrayList<Object> message = new ArrayList();
+            if (response.getStatus() == 200) {
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/auth/OpenStackAuthenticationToken.java]: Magic number

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/auth/OpenStackAuthenticationToken.java [hunk 2]
```diff
@@ -109,200 +111,61 @@ public void initialize(ArrayList<Object> params) {
...
-        try {
-            response = httpClient.execute(postRequest);
+                if (jsonObject.containsKey("token")) {
 
-            localDate = new Date();
+                    JSONObject tokenObject = (JSONObject) jsonObject.get("token");
+                    token = (String) tokenObject.get("id");
+                    tenantId = (String) ((JSONObject) tokenObject.get("tenant")).get("id");
 
-            if ((response.getStatusLine().getStatusCode() != 201) && (response.getStatusLine().getStatusCode() != 200)) {
```
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/auth/OpenStackAuthenticationToken.java]: magic numbers

File: rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/auth/OpenStackAuthenticationToken.java (hunk 1 above)
Comment (flopezag) [inline rest-api/src/main/java/com/telefonica/euro_iaas/sdc/rest/auth/OpenStackAuthenticationToken.java]: What's about 201?

**Comment by flopezag** [comment]
Coverage decreased... just add new Unit Test

**Comment by flopezag** [comment]
LGTM

**Pull request merged**